app/src/main/java/ch/epfl/sdp/appart/login/FirebaseLoginService.java

```diff
@@ -130,22 +123,43 @@ private FirebaseUser getCurrentFirebaseUser() {
         return user;
     }
 
-
-    private <T> void addCallbackToTask(Task<T> task, LoginCallback callback) {
-        task.addOnCompleteListener(new LoginOnCompleteListener<>(callback));
+    /**
+     * Converts an AuthResult to a User
+     *
+     * @param result the AuthResult coming from Firebase
+     * @return a user
+     */
+    private User getUserFromAuthResult(AuthResult result) {
+        FirebaseUser user = getCurrentFirebaseUser();
+        return new AppUser(user.getUid(), user.getEmail());
     }
 
-    private static class LoginOnCompleteListener<T> implements OnCompleteListener<T> {
-        private final LoginCallback callback;
-
-        LoginOnCompleteListener(LoginCallback callback) {
-            this.callback = callback;
-        }
-
+    /**
+     * Interface that handles the conversion of a task from one type to another
+     *
+     * @param <FROM> the type we want to convert
+     * @param <TO>   the type we want to get
+     */
+    private interface ConvertTask<FROM, TO> {

```

Comment:
TaskToFuturAdapter ?

Reply:
I don't think of it like that because it's more a converter than an adapter. We do not wrap a type in another but rather transform a type into another. By the way, this actually should be called ConvertType or TypeAdapter because in itself it has nothing to do with tasks.